File: web/src/plugin/image/api/image.js
import service from '@/utils/request'

export const createImage = (data) => {
  return service({
    url: '/image/createImage',
    method: 'post',
    data
  })
}

export const deleteImage = (params) => {
  return service({
    url: '/image/deleteImage',
    method: 'delete',
    params
  })
}

export const deleteImageByIds = (params) => {
  return service({
    url: '/image/deleteImageByIds',
    method: 'delete',
    params
  })
}

export const updateImage = (data) => {
  return service({
    url: '/image/updateImage',
    method: 'put',
    data
  })
}

export const findImage = (params) => {
  return service({
    url: '/image/findImage',
    method: 'get',
    params
  })
}

export const getImageList = (params) => {
  return service({
    url: '/image/getImageList',
    method: 'get',
    params
  })
}

File: server/plugin/image/router/image.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Image = new(image)

type image struct{}

func (r *image) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("image").Use(middleware.OperationRecord())
		group.POST("createImage", apiImage.CreateImage)
		group.DELETE("deleteImage", apiImage.DeleteImage)
		group.DELETE("deleteImageByIds", apiImage.DeleteImageByIds)
		group.PUT("updateImage", apiImage.UpdateImage)
	}
	{
		group := private.Group("image")
		group.GET("findImage", apiImage.FindImage)
		group.GET("getImageList", apiImage.GetImageList)
	}
}
